.github/workflows/build-on-pr.yml

```diff
@@ -0,0 +1,24 @@
+name: Build on PR
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - uses: actions/setup-go@v4
+      with:
+        go-version: '1.20'
```

Comment:
Is it possible to use go version latest here? There are some good upgrades with each version and they're backwards compatible

Reply:
of course but we should start by bumping, go.mod, dockerfile, and other things before we bump it here